core/kernel/trace_ext.c

```diff
@@ -21,14 +22,18 @@ void __weak plat_trace_ext_puts(const char *str __unused)
 
 void trace_ext_puts(const char *str)
 {
-	uint32_t itr_status = thread_mask_exceptions(THREAD_EXCP_ALL);
+	uint32_t itr_status = 0;
 	bool mmu_enabled = cpu_mmu_enabled();
 	bool was_contended = false;
-	const char *p;
+	const char *p = NULL;
 
-	if (mmu_enabled && !cpu_spin_trylock(&puts_lock)) {
-		was_contended = true;
-		cpu_spin_lock_no_dldetect(&puts_lock);
+	if (!IS_ENABLED(CFG_TEE_CONSOLE_UNLOCKED)) {
+		itr_status = thread_mask_exceptions(THREAD_EXCP_ALL);
+
+		if (mmu_enabled && !cpu_spin_trylock(&puts_lock)) {
```

Comment:
You can still loop around `cpu_spin_trylock()` and unmask and mask exceptions as needed to take the spinlock, this should avoid mixed logs while still not keeping interrupts masked for a longer time.